libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1199,6 +1199,29 @@ void CompilerUtils::computeHashStatic()
 	m_context << u256(32) << u256(0) << Instruction::KECCAK256;
 }
 
+void CompilerUtils::copyContractCodeToMemory(ContractDefinition const& contract, bool _creation)
+{
+	string which = _creation ? "Creation" : "Runtime";
+	m_context.callLowLevelFunction(
+		"$copyContract" + which + "CodeToMemory_" + contract.type()->identifier(),
+		1,
+		1,
+		[&contract, _creation](CompilerContext& _context)
+		{
+			// copy the contract's code into memory
+			eth::Assembly const& assembly =
+				_creation ?
+				_context.compiledContract(contract) :
+				_context.compiledContractRuntime(contract);
+			// pushes size
+			auto subroutine = _context.addSubroutine(make_shared<eth::Assembly>(assembly));
```

Comment:
This looks like it would be VERY VERY bad - it creates a full copy of the assembly. This means we run the optimizer and assembler multiple times and in case it creates any different result, we will get different bytecode here.